### PR TITLE
Fixed product image URL retrieval to ensure it returns the correct small image URL or an empty string if no image is available.

### PR DESCRIPTION
**Pull request by ahmardi**
…all image URL or an empty string if no image is available.

### CODE REVIEW

**Review by sameh-amwal — APPROVED** [review]

File: Model/Button/GetCartButtonConfig.php [hunk 1]
```diff
@@ -274,9 +274,21 @@ private function getOrderContent($quote): array
...
         return $orderContent;
     }
+
+    /**
+     * @param $product
+     * @return string
+     */
+    private function getProductImageUrl($product)
+    {
+        if (!$product->getData('small_image')) {
```
Comment (Skullsneeze) [inline Model/Button/GetCartButtonConfig.php:289]: If we are using the image roles, maybe we can add an additional fallback to the "thumbnail" role so we have an additional option?

File: Model/Button/GetCartButtonConfig.php [hunk 2]
```diff
@@ -274,9 +274,21 @@ private function getOrderContent($quote): array
...
             ];
         }
         return $orderContent;
     }
+
+    /**
+     * @param $product
+     * @return string
+     */
+    private function getProductImageUrl($product)
```
Comment (Skullsneeze) [inline Model/Button/GetCartButtonConfig.php:287]: I think we can add type hints here (and in the docblock)
```suggestion
    private function getProductImageUrl(ProductInterface $product): string
```

**Review by Skullsneeze — CHANGES_REQUESTED** [review]

File: Model/Button/GetCartButtonConfig.php (hunk 1 above)
Comment (ahmardi) [reply Model/Button/GetCartButtonConfig.php:289]: I'll try this

File: Model/Button/GetCartButtonConfig.php [hunk 3]
```diff
@@ -274,9 +275,22 @@ private function getOrderContent($quote): array
...
     }
+
+    /**
+     * @param ProductInterface $product
+     * @return string
+     */
+    private function getProductImageUrl(ProductInterface $product): string
+    {
+        $image = $product->getData('small_image') ?: $product->getData('thumbnail');
+        if(!$image) {
```
Comment (Skullsneeze) [inline Model/Button/GetCartButtonConfig.php:291]:
```suggestion
        if (!$image) {
```

**Pull request merged**